Point2d  robot_pos
int32    which_task
